tests/test_factorgraph.py

```diff
@@ -904,3 +904,103 @@ def test_and_rounding_error_simple():
     bp.bp_loopy(5, initialize_states=False)
     assert (bp.get_distribution("K0") >= 0.0).all()
     assert (bp.get_distribution("K1") >= 0.0).all()
+
+
+def test_dampening_nochange():
+    graph = """
+        NC 2
+        PROPERTY s1: x = a^b
+        VAR MULTI x
+        VAR MULTI a
+        VAR MULTI b
+        """
+    distri_a = np.array([[0.6, 0.4]])
+    distri_b = np.array([[0.8, 0.2]])
+    graph = FactorGraph(graph)
+    n = 1
+    bp_state_no_dampen = BPState(graph, n)
+    bp_state_no_dampen.set_evidence("a", distri_a)
+    bp_state_no_dampen.set_evidence("b", distri_b)
+    bp_state_full_dampen = BPState(graph, n)
+    bp_state_full_dampen.set_evidence("a", distri_a)
+    bp_state_full_dampen.set_evidence("b", distri_b)
+
+    bp_state_no_dampen.bp_loopy(5, initialize_states=False)
+    bp_state_full_dampen.bp_loopy(5, initialize_states=False, alpha=1.0)
```

Comment:
I am not sure to understand the purpose of this test. `alpha=1.0` is on the border of admissible (at least alpha=1.0 seems a bit worthless to use in practice). There should be a comment explaining why this is expected to work.